src/parser/index.ts

```diff
@@ -145,31 +145,46 @@ export class Parser {
 
     /**
      * Simplify the logical expression if possible.
+     * - AND, OR with one operand is collapsed.
+     * - Logical expressions without operands are treated as collection.
+     * @param collapsible
      * @param operands
      */
-    const logicalExpressionReducer = (operands: Evaluable[]): Evaluable | undefined =>
-      operands.length === 1 ? operands[0] : undefined
+    const logicalExpressionReducer = (
+      collapsible: boolean,
```

Comment:
Why not make `collapsible` the second argument and optional with a default of `false`?

Reply:
219b9f5a280fff7c8719b3b952ed3a352278f47a